src/access/tests/TestAccessControl.cpp

```diff
@@ -28,497 +29,980 @@ namespace {
+    }
+    for (size_t i = 0; i < entryData.GetTargetCount(); ++i)
+    {
+        ReturnErrorOnFailure(entry.AddTarget(nullptr, entryData.targets[i]));
+    }
+    return CHIP_NO_ERROR;
 }
 
-TestTarget Target(EndpointId endpoint)
+CHIP_ERROR ClearAccessControl(AccessControl & ac)
 {
-    return { .flags = TestTarget::kEndpoint, .endpoint = endpoint };
+    CHIP_ERROR err;
+    do
+    {
+        err = accessControl.DeleteEntry(0);
+    } while (err == CHIP_NO_ERROR);
+    return CHIP_NO_ERROR;
 }
 
-TestTarget Target(ClusterId cluster)
+CHIP_ERROR CompareAccessControl(AccessControl & ac, const EntryData * entryData, size_t count)
 {
-    return { .flags = TestTarget::kCluster, .cluster = cluster };
+    Entry entry;
+    for (size_t i = 0; i < count; ++i, ++entryData)
+    {
+        ReturnErrorOnFailure(ac.ReadEntry(i, entry));
+        ReturnErrorOnFailure(CompareEntry(entry, *entryData));
+    }
+    ReturnErrorCodeIf(ac.ReadEntry(count, entry) == CHIP_NO_ERROR, CHIP_ERROR_INCORRECT_STATE);
+    return CHIP_NO_ERROR;
 }
 
-struct TestEntryDelegate
+CHIP_ERROR LoadAccessControl(AccessControl & ac, const EntryData * entryData, size_t count)
 {
-    FabricIndex fabricIndex = 0;
-    AuthMode authMode       = AuthMode::kNone;
-    Privilege privilege     = Privilege::kView;
-    SubjectId subjects[kSubjectsPerEntry + 1];
-    TestTarget targets[kTargetsPerEntry + 1];
-    const char * tag = "";
-    bool touched     = false;
+    Entry entry;
+    for (size_t i = 0; i < count; ++i, ++entryData)
+    {
+        ReturnErrorOnFailure(ac.PrepareEntry(entry));
+        ReturnErrorOnFailure(LoadEntry(entry, *entryData));
+        ReturnErrorOnFailure(ac.CreateEntry(nullptr, entry));
+    }
+    return CHIP_NO_ERROR;
+}
+
+constexpr EntryData entryData1[] =
+{
+    { .fabricIndex = 1, .privilege = Privilege::kAdminister, .authMode = AuthMode::kCase,
+        .subjects = { 0x1111111111111111 }, },
+    { .fabricIndex = 1, .privilege = Privilege::kView, .authMode = AuthMode::kCase, },
+    { .fabricIndex = 2, .privilege = Privilege::kAdminister, .authMode = AuthMode::kCase,
+        .subjects = { 0x2222222222222222 }, },
+    { .fabricIndex = 1, .privilege = Privilege::kOperate, .authMode = AuthMode::kCase,
+        .targets = { { .flags = Target::kCluster, .cluster = kOnOffCluster } }, },
+    { .fabricIndex = 2, .privilege = Privilege::kManage, .authMode = AuthMode::kPase,
+        .subjects = { kPaseVerifier1 },
+        .targets = { { .flags = Target::kCluster | Target::kEndpoint, .cluster = kOnOffCluster, .endpoint = 2 } }, },
+    { .fabricIndex = 2, .privilege = Privilege::kProxyView, .authMode = AuthMode::kGroup,
+        .subjects = { kGroup2 },
+        .targets = { { .flags = Target::kCluster | Target::kEndpoint, .cluster = kLevelControlCluster, .endpoint = 1 }, { .flags = Target::kCluster, .cluster = kOnOffCluster }, { .flags = Target::kEndpoint, .endpoint = 2 } }, },
 };
 
-TestEntryDelegate entries[] = {
+struct CheckData
```

Comment:
A better name would be `TestVectorData`

Reply:
This is the data for check. The other data contains entries. Everything here is "test", and I'm not sure what you mean by "vector" so I don't think that makes the name clearer.